boards/arm/sama5/giant-board/src/sam_ethernet.c

```diff
@@ -0,0 +1,328 @@
+/****************************************************************************
+ *  boards/arm/sama5/giant-board/src/sam_ethernet.c
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one or more
+ *  contributor license agreements.  See the NOTICE file distributed with
+ *  this work for additional information regarding copyright ownership.  The
+ *  ASF licenses this file to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance with the
+ *  License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+/* Force verbose debug on in this file only to support unit-level testing. */
+
+#ifdef CONFIG_NETDEV_PHY_DEBUG
+#  undef  CONFIG_DEBUG_INFO
+#  define CONFIG_DEBUG_INFO 1
+#  undef  CONFIG_DEBUG_NET
+#  define CONFIG_DEBUG_NET 1
+#endif
+
+#include <string.h>
+#include <assert.h>
+#include <debug.h>
+
+#include <nuttx/irq.h>
+#include <nuttx/arch.h>
+
+#include "sam_pio.h"
+#include "sam_ethernet.h"
+
+#include "giant-board.h"
+
+#ifdef HAVE_NETWORK
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef CONFIG_SAMA5_EMACA
+#  undef CONFIG_SAMA5_EMAC_ISETH0
+#endif
+
+#ifdef CONFIG_SAMA5_EMAC_ISETH0
+#  define SAMA5_EMAC_DEVNAME "eth0"
+#  define SAMA5_GMAC_DEVNAME "eth1"
+#else
```

Comment:
Remove all references to GMAC.  See below.

Reply:
@patacongo Will correct, if the peripheral is an EMAC.

Reply:
In NuttX, the 10/100mbit GMAC peripheral is referred to as EMAC.

That is because the same code is used for the entire family.  The SAMAD3 is the only family to have a gigabit GMAC and a 10/100 mbit EMAC.  The SAMAD2 and 4 and got peripherals called GMAC, but they are the same as the D3 EMAC, not the GMAC.  So in order to share the code, they have to be called EMAC.

But this is mute.  You need to remove all Ethernet support anyway.  The Giant Board does not bring other the Ethernet so you can't have any kind of Ethernet on that board (or NAND or buttons).

Let's not put unused and unusable files in the port.  If, in the future, someone hacks the Giant Board to support Ethernet, then we can bring the correct driver in (the one there now is not correct).